### DISCUSSION TITLE
Migration scripts?

### DISCUSSION THREAD

**Discussion by christian-oudard** [diesel-rs/diesel#2726] [Q&A]
So, I am doing a migration which needs to port data to the new schema. The query to do that looks something like this:
```
UPDATE table SET field = q.otherfield
FROM (
    SELECT t2.some_id, t2.otherfield FROM t1 JOIN t2 ON t1.id = t2.id
    WHERE condition
) AS q
WHERE some_id = t2.some_id
```

Unfortunately, we have to deploy this on an environment using SQLite 3.22, which doesn't support `UPDATE FROM`. The only way to do this data manipulation is with multiple queries, which would require a script to connect the inputs of one query to the outputs of another query.
```
let data = SELECT ...
for line in data {
    UPDATE ...
}
```

Is there a way to do scripted migrations in diesel? Does `barrel` support something like this?

**Comment by weiznich** [comment]
That would require access to the full `Connection` trait instead of only to the `SimpleConnection` trait inside of [`diesel::migration::Migration::run`](https://docs.diesel.rs/1.4.x/diesel/migration/trait.Migration.html#tymethod.run). That's not supported on any release yet, see https://github.com/diesel-rs/diesel/issues/2457 for details. 
That written #2657 will likely allow to write such migrations in a future version of diesel.

**Reply by christian-oudard** [reply]
I think we're just going to have to use a recent version of SQLite, rather than support the old version.
